WordPressKit/WordPressKit/AccountServiceRemoteREST.m

```diff
@@ -193,13 +193,47 @@ - (void)requestWPComAuthLinkForEmail:(NSString *)email
                             clientID:(NSString *)clientID
                         clientSecret:(NSString *)clientSecret
                          wpcomScheme:(NSString *)scheme
-                             success:(void (^)(void))success failure:(void (^)(NSError *error))failure
+                             success:(void (^)(void))success
+                             failure:(void (^)(NSError *error))failure
 {
-    NSAssert([email length] > 0, @"Needs an email address.");
+    [self requestWPComLinkForEmail:email
+                      endpointPath:@"auth/send-login-email"
+                          clientID:clientID
+                      clientSecret:clientSecret
+                       wpcomScheme:scheme
+                           success:success
+                           failure:failure];
+}
 
-    NSString *path = [self pathForEndpoint:@"auth/send-login-email"
-                                     withVersion:ServiceRemoteWordPressComRESTApiVersion_1_1];
+- (void)requestWPComSignupLinkForEmail:(NSString *)email
+                              clientID:(NSString *)clientID
+                          clientSecret:(NSString *)clientSecret
+                           wpcomScheme:(NSString *)scheme
+                               success:(void (^)(void))success
+                               failure:(void (^)(NSError *error))failure
+{
+    [self requestWPComLinkForEmail:email
+                      endpointPath:@"auth/send-signup-email"
+                          clientID:clientID
+                      clientSecret:clientSecret
+                       wpcomScheme:scheme
+                           success:success
+                           failure:failure];
+}
 
+- (void)requestWPComLinkForEmail:(NSString *)email
+                    endpointPath:(NSString *)endpointPath
+                        clientID:(NSString *)clientID
+                    clientSecret:(NSString *)clientSecret
+                     wpcomScheme:(NSString *)scheme
+                         success:(void (^)(void))success
+                         failure:(void (^)(NSError *error))failure
+{
+    NSAssert([email length] > 0, @"Needs an email address.");
+    
+    NSString *path = [self pathForEndpoint:endpointPath
```

Comment:
ok this one is theoretical, but afaik there's nothing forcing both magic link methods to be on the same version of the API. If that's right, it seems like the call to `pathForEndpoint: withVersion:` should be in `requestWPComAuthLinkForEmail` and `requestWPComSignupLinkForEmail`.

I know this is pedantic. What do you think?

Reply:
I actually don't think it's pedantic - that didn't occur to me. Good point!